packages/library/src/commonMain/kotlin/io/realm/Realm.kt

```diff
@@ -48,82 +34,68 @@ class Realm private constructor(configuration: RealmConfiguration, dbPointer: Na
-    }
-
-    fun beginTransaction() {
-        RealmInterop.realm_begin_write(dbPointer!!)
-    }
-
-    fun commitTransaction() {
-        RealmInterop.realm_commit(dbPointer!!)
-    }
-
-    fun cancelTransaction() {
-        TODO()
-    }
+    /**
+     * Open a Realm instance. This instance grants access to an underlying Realm file defined by
+     * the provided [RealmConfiguration].
+     *
+     * FIXME Figure out how to describe the constructor better
+     * FIXME Once the implementation of this class moves to the frozen architecture
+     *  this constructor should be the primary way to open Realms (as you only need
+     *  to do it once pr. app).
+     */
+    public constructor(configuration: RealmConfiguration) :
+        this(configuration, RealmInterop.realm_open(configuration.nativeConfig))
 
-    fun <T : RealmObject> create(type: KClass<T>): T {
-        return io.realm.internal.create(configuration.mediator, dbPointer!!, type)
+    /**
+     * TODO Add docs when this method is implemeted
+     */
+    suspend fun <R> write(function: MutableRealm.() -> R): R {
+        TODO("Awaiting implementation of the Frozen Architecture")
     }
-    // Convenience inline method for the above to skip KClass argument
-    inline fun <reified T : RealmObject> create(): T { return create(T::class) }
 
     /**
-     * Creates a copy of an object in the Realm.
+     * NOTE Avoid calling this method on the UI thread, instead use [Realm.write].
+     *
+     * Modify the underlying Realm file by creating a write transaction on the current thread. Write
+     * transactions automatically commit any changes made when the closure returns unless
+     * [MutableRealm.cancelWrite] was called.
      *
-     * This will create a copy of an object and all it's children. Any already managed objects will
-     * not be copied, including the root `instance`. So invoking this with an already managed
-     * object is a no-operation.
+     * The write transaction always represent the latest version of data in the Realm file, even if the calling
+     * Realm not yet represent this.
      *
-     * @param instance The object to create a copy from.
-     * @return The managed version of the `instance`.
+     * TODO Better explanation here.
+     * @return any value returned from the provided write function.
      */
-    fun <T : RealmObject> copyToRealm(instance: T): T {
-        return copyToRealm(configuration.mediator, dbPointer!!, instance)
-    }
-
-    fun <T : RealmObject> objects(clazz: KClass<T>): RealmResults<T> {
-        return RealmResults(
-            dbPointer!!,
-            @Suppress("SpreadOperator") // TODO PERFORMANCE Spread operator triggers detekt
-            { RealmInterop.realm_query_parse(dbPointer!!, clazz.simpleName!!, "TRUEPREDICATE") },
-            clazz,
-            configuration.mediator
-        )
+    @Suppress("TooGenericExceptionCaught")
+    fun <R> writeBlocking(function: MutableRealm.() -> R): R {
```

Comment:
As in `writeBlock` instead? I was going with the pattern set in the Kotlin API like `runBlocking { }` ... IMO that consistency is worth more than saving a few characters especially since the IDE autocomplete it. But I can see it being annoying to read if you do this a lot... I also had the assumption that in real-world apps `realm.write { }` is going to be the predominant use case, so optimizing this name was not worthwhile.  

Thoughts?

Reply:
Ahh. Sorry. I was just referring to the argument. `function`->`block`. 

`writeBlocking` is fine as, I agree that, `write` would be the predominant use case.

Reply:
Ah okay, I don't have any strong preference there, as long as we use the same word consistently. `block` seems fine to me 👍